src/model/model.ts

```diff
@@ -1,43 +1,105 @@
+  };
 
   constructor(data: any) {
-    this.data = data;
+    for (const key in data) {
+      this[key] = data[key];
+    }
   }
 
-  save() {
-    validateSchema(this.data, this.__schema);
-    return save(generateUUID(), this.data, this.__collection);
+  /**
+   * return id value
+   */
+  getId(): string {
+    return this[this.$.ID_KEY];
   }
 
-  static find = (params): Promise<any> => Promise.resolve();
-  static create = (data: any): Promise<any> => Promise.resolve();
-}
-
-export const model = (name, schema?, connection?, options?) => {
-  const collection = getCollection(name);
-  return class ModelFactory extends Model {
-    constructor(data) {
-      super(data);
-      this.__collection = collection;
-      this.__schema = schema;
+  /**
+   * Save or Update document, I
+   * If the document has id (key) will be update
+   * Else a new id (key) will be assigned and document will be stored
+   */
+  async save() {
+    const data = extractDataFromModel(this);
+    const options: any = {};
+    const id = this.getId();
+    if (!id) {
+      data[this.$.ID_KEY] = generateUUID();
+    } else {
+      const { cas } = await this.$.collection.get(id);
+      if (cas) {
+        options.cas = cas;
+      }
     }
+    const addedMetadata = { ...data, [COLLECTION_KEY]: this.$.collectionName };
+    return storeLifeCycle({ data: addedMetadata, options, model: this.$ });
+  }
 
-    static find = (params): Promise<any> => {
-      return find(params, collection);
-    };
+  /**
+   * Remove document from BD
+   */
+  remove(options = {}) {
+    return removeLifeCicle({ id: this.getId(), options, model: this.$ });
+  }
 
-    static create = (data: Record<string, any>): Promise<any> => save(generateUUID(), data, collection);
+  /**
+   * Return javascript object with data
+   */
+  toCoo() {
+    return extractDataFromModel(this);
+  }
 
-    static insertMany(data: Record<string, any>[]): Promise<any> {
-      return Promise.all(data.map((d) => this.create(d)));
-    }
-  };
-};
+  /**
+   * Return javascript object serialized to JSON
+   */
+  toJSON() {
+    return JSON.stringify(this.toCoo());
+  }
+
+  /**
+   * Find documents base on filter on the collection
+   */
+  static find = (params): Promise<any> => Promise.resolve(params);
+
+  /**
+   * Add new document to collection
+   */
+  static create = (data: any): Promise<any> => Promise.resolve(data);
+
+  /**
+   * Update document in the collection
+   * Can update document partial or complete
+   * id parameter will have more priority that data.id
+   */
+  static update = (data: any, id?: string): Promise<any> => Promise.resolve({ data, id });
+
+  /**
+   * Replace document in the collection
+   * Will replace the entire document with the given data
+   * id parameter will have more priority that data.id
+   */
+  static replace = (data: any, id?: string): Promise<any> => Promise.resolve({ data, id });
+
+  /**
+   * Remove document in the collection with the provide id
```

Comment:
Should be: `Remove document in the collection with the (provided) id`

Reply:
fixed the typo